tests/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,229 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/evmos/evmos/v10/tests/e2e/upgrade"
+)
+
+const (
+	localRepository = "evmos"
+	localVersionTag = "local"
+
+	defaultChainID = "evmos_9000-1"
+
+	defaultManagerNetwork = "evmos-local"
+
+	firstUpgradeHeight = 50
+)
+
+var (
+	// common
+	maxRetries = 10 // max retries for json unmarshalling
+)
+
+type upgradeParams struct {
+	InitialVersion string
+	TargetVersion  string
+	MountPath      string
+
+	MigrateGenesis bool
+	SkipCleanup    bool
+}
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	tmpDirs        []string
+	upgradeManager *upgrade.Manager
+	hermesResource *dockertest.Resource
+	upgradeParams  upgradeParams
+}
+
+type status struct {
+	LatestHeight string `json:"latest_block_height"`
+}
+
+type syncInfo struct {
+	SyncInfo status `json:"SyncInfo"`
+}
+
+func TestIntegrationTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up e2e integration test suite...")
+	var err error
+
+	s.loadUpgradeParams()
+
+	s.upgradeManager, err = upgrade.NewManager(defaultManagerNetwork)
+	s.Require().NoError(err, "upgrade manager creation error")
+}
+
+func (s *IntegrationTestSuite) loadUpgradeParams() {
+	preV := os.Getenv("INITIAL_VERSION")
+	if preV == "" {
+		s.Fail("no pre-upgrade version specified")
+	}
+	s.upgradeParams.InitialVersion = preV
+
+	postV := os.Getenv("TARGET_VERSION")
+	if postV == "" {
+		s.Fail("no post-upgrade version specified")
+	}
+	s.upgradeParams.TargetVersion = postV
+
+	migrateGenFlag := os.Getenv("MIGRATE_GENESIS")
+	migrateGenesis, err := strconv.ParseBool(migrateGenFlag)
+	s.Require().NoError(err, "invalid migrate genesis flag")
+	s.upgradeParams.MigrateGenesis = migrateGenesis
+
+	skipFlag := os.Getenv("E2E_SKIP_CLEANUP")
+	skipCleanup, err := strconv.ParseBool(skipFlag)
+	s.Require().NoError(err, "invalid skip cleanup flag")
+	s.upgradeParams.SkipCleanup = skipCleanup
+
+	mountPath := os.Getenv("MOUNT_PATH")
+	if postV == "" {
+		s.Fail("no mount path specified")
+	}
+	s.upgradeParams.MountPath = mountPath
+	s.T().Log("upgrade params: ", s.upgradeParams)
+}
+
+func (s *IntegrationTestSuite) runInitialNode() {
+	node := upgrade.NewNode(localRepository, s.upgradeParams.InitialVersion)
+	err := s.upgradeManager.RunNode(node)
+	s.Require().NoError(err, "can't run initial node")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// wait untill node starts and produce some blocks
+	err = s.upgradeManager.WaitForHeight(ctx, 5)
+	s.Require().NoError(err)
+
+	s.T().Logf("successfully started initial node version: [%s]", s.upgradeParams.InitialVersion)
+}
+
+func (s *IntegrationTestSuite) proposeUpgrade() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
nit, would leave empty space after defer for readability. 
```suggestion
	defer cancel()
	
```